token-handler/src/mint_tokens.rs

```diff
@@ -1,79 +1,205 @@
-use multiversx_sc::api::ESDT_NFT_CREATE_FUNC_NAME;
-use multiversx_sc::types::ManagedArgBuffer;
-use multiversx_sc::types::{ManagedVec, MultiValueEncoded};
+use multiversx_sc::api::{ESDT_MULTI_TRANSFER_FUNC_NAME, ESDT_NFT_CREATE_FUNC_NAME};
+use multiversx_sc::types::{
+    system_proxy, EsdtTokenPayment, ManagedArgBuffer, MultiValueEncoded, ToSelf,
+};
+use multiversx_sc::types::{ManagedVec, TokenIdentifier};
 use multiversx_sc::{codec, err_msg};
-use transaction::OperationEsdtPayment;
+use transaction::{
+    GasLimit, OperationData, OperationEsdtPayment, OperationTuple,
+    StolenFromFrameworkEsdtTokenData, TransferData,
+};
 
-use crate::common;
+const TRANSACTION_GAS: GasLimit = 30_000_000;
+
+use crate::{burn_tokens, common};
 
 #[multiversx_sc::module]
-pub trait MintTokens: utils::UtilsModule + common::storage::CommonStorage {
-    #[endpoint(mintTokens)]
-    fn mint_tokens(&self, operation_tokens: MultiValueEncoded<OperationEsdtPayment<Self::Api>>) {
+pub trait TransferTokensModule:
+    utils::UtilsModule
+    + common::storage::CommonStorage
+    + common::events::EventsModule
+    + burn_tokens::BurnTokensModule
+    + tx_batch_module::TxBatchModule
+{
+    #[payable("*")]
+    #[endpoint(transferTokens)]
+    fn transfer_tokens(
```

Comment:
We will also need the `original_sender` (operation.data.op_sender) in argument. This will be later used for transfer and execute by user functionality